packages/components/src/components/sequence/ReturnLabel.vue

```diff
@@ -10,21 +11,53 @@ import { ActionSpec } from './ActionSpec';
 export default {
   name: 'v-sequence-return-label',
 
-  components: {},
+  data() {
+    return {
+      hover: false,
+    };
+  },
 
   props: {
     actionSpec: {
       type: ActionSpec,
       required: true,
       readonly: true,
     },
+    appMap: {
+      type: Object,
+      required: true,
+      readonly: true,
+    },
   },
 
   computed: {
     label() {
       const { nodeResult } = this.actionSpec;
       return !nodeResult || nodeResult === 'void' ? '' : nodeResult;
     },
+
+    value() {
+      if (!this.actionSpec.eventIds || !this.appMap || !this.appMap.events) {
+        return false;
+      }
+      const eventId = this.actionSpec.eventIds[0];
+      const event = this.appMap.events.find(
+        (e) => e.parent_id && e.event == 'return' && e.parent_id === eventId
+      );
```

Comment:
You can retrieve the event without iterating by accessing the event via the `eventsById` map. E.g.:
```ts
this.appMap.eventsById[eventId]
```
This will give you the call event. You can get the return value via the `returnValue` accessor. If you need the return event, `returnEvent` is also available.

Reply:
Updated.